src/components/form.jsx

```diff
@@ -1,48 +1,68 @@
 import { useState, useEffect } from "react";
+import { LoadingSpinner } from "./loadingSpinner";
 
 const Form = () => {
   const [data, setData] = useState(null);
   const [error, setError] = useState(false);
   const [errorMessage, setErrorMessage] = useState(false);
   const [userInput, setUserInput] = useState("");
   const [isUserInputBlank, setIsUserInputBlank] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  function clearDataBeforeFetch() {
+    setErrorMessage("");
+    setData(false);
+    setIsUserInputBlank(false);
+    setError(false);
+  }
 
   const fetchData = (e) => {
+    clearDataBeforeFetch();
```

Comment:
Maybe we should add a check here to not fetch if the userInput hasn't changed? If so don't fetch, cause right now it's giving a weird glitch with the loader. Otherwise we can address this in a different issue

Reply:
I'm going to take a look at it tomorrow!

Reply:
The spinner works okay for me! Take a look at this @ctoffanin and we can go over this PR again.

Reply:
I've updated the PR. Give it a check @adropofilm

Reply:
Just checked it out, LGTM :D @ctoffanin @Njong392

Reply:
I think I got the problem mentioned by Emy.
I did an update. Can you both check if everything makes sense @adropofilm @Njong392 ?

Reply:
> I think I got the problem mentioned by Emy. I did an update. Can you both check if everything makes sense @adropofilm @Njong392 ?

You're a star @ctoffanin I think this draft is ready to actually be opened for merging! Thanks so much for the new changes.

Reply:
I didn't turn it into a mergeable PR because I noticed this when removing the delay.

https://user-images.githubusercontent.com/1682188/200365453-8d70cb9a-def5-4d8e-9fe6-cdb707dd122c.mov

I reduced the delay to 300ms, so you can take a look at it and decide what you want to do. 
Either that fast flash of loader you see in the footage or what it's now in the PR.
@adropofilm @Njong392 Let me know what route you want to go.

Reply:
uhm... I think you can take out the delay entirely. The delay is usually just to simulate a situation where the user might have a slow internet connection. I've heard people say they intentionally put delays (I don't know why). But i know that in production code, the setTimeOut function is not there. Thoughts @adropofilm ?

Reply:
So you're ok to with the behaviour you see in the footage above, that small glimpse of the spinner?
I do agree the delay should not be there. I wanted to show an option anyway.

Reply:
Oh no. I'm saying we should remove the delay altogether.

Reply:
> Oh no. I'm saying we should remove the delay altogether.

Yeah, I agree.

Reply:
Awesome. You can remove the delay, and then we'll get this ready to merge if all else is fine :+1:

Reply:
@adropofilm we waiting on you to take a look at this before we merge.

Reply:
Done! Looks good to me, great job @ctoffanin <3